java-core/google-cloud-core/src/main/java/com/google/cloud/testing/BaseEmulatorHelper.java

```diff
@@ -112,14 +115,21 @@ protected final void startProcess(String blockUntilOutput)
     }
   }
 
+  /** This method is obsolete. Use {@link #waitForProcessDuration(java.time.Duration)} instead */
+  @ObsoleteApi("Use waitForProcessDuration() instead")
+  protected final int waitForProcess(org.threeten.bp.Duration timeout)
+      throws IOException, InterruptedException, TimeoutException {
+    return waitForProcessDuration(toJavaTimeDuration(timeout));
+  }
+
   /**
    * Waits for the local service's subprocess to terminate, and stop any possible thread listening
    * for its output.
    */
-  protected final int waitForProcess(Duration timeout)
+  protected final int waitForProcessDuration(java.time.Duration timeout)
       throws IOException, InterruptedException, TimeoutException {
     if (activeRunner != null) {
-      int exitCode = activeRunner.waitFor(timeout);
+      int exitCode = activeRunner.waitForDuration(timeout);
       activeRunner = null;
       return exitCode;
     }
```

Comment:
Can we double check that this overload is needed? I believe it's protected and final and subclasses shouldn't be able to override this method. I think if we just migrate all internal uses cases in this repo to use java.time.Duration, we should be good.

Reply:
Thanks for checking this. Indeed there were a couple of unnecessary overloads. I switched this one directly to java.time. CLIRR still found it as a incompatible change, so I added a new clirr ignore entry.

https://github.com/googleapis/sdk-platform-java/blob/2d7950f445c93c77f0999caf190422635e50f1dd/java-core/google-cloud-core/clirr-ignored-differences.xml#L21-L27

Reply:
Oh, looks like the datastore folks [use this method directly](https://github.com/googleapis/java-datastore/blob/92952c8e033aaeed7f0dcded1226bbaed577e140/google-cloud-datastore/src/main/java/com/google/cloud/datastore/testing/LocalDatastoreHelper.java#L322-L326), so me may want the overload for the time being? (ci [check failure](https://github.com/googleapis/sdk-platform-java/actions/runs/11671715727/job/32498740618?pr=3330))

Reply:
I restored the overload

Reply:
Ok, right I believe that makes sense. They can't override this method but they can extend from BaseEmulatorHelper and use the parent implementation. Thanks for checking!